manim/utils/polylabel.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+from __future__ import annotations
+
+from queue import PriorityQueue
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from manim.typing import Point3D
+
+
+class Polygon:
+    def __init__(self, rings: list[np.ndarray]) -> None:
```

Comment:
We've discussed this earlier when we talked about `np.ndarray`, `Point3D` and a possible new `PointND` type alias. There are two approaches to this:
- Continue with the current approach for typing `rings`. In this case, in Manim:
  - Instead of `list`, we use `Sequence`, because this prevents issues when overriding classes.
  - Instead of `np.ndarray`, we use `npt.NDArray[<some_type>]`, where `npt` comes from `import numpy.typing as npt`. In this case, `<some_type>` would be `manim.typing.ManimFloat`.
- Instead of the verbose `Sequence[npt.NDArray[ManimFloat]]`, my suggestion would be simply adding new type aliases: `PointND` and `PointND_Array`.